api/v1alpha1/datadogagent_validation.go

```diff
@@ -39,6 +39,13 @@ func IsValidDatadogAgent(spec *DatadogAgentSpec) error {
 		}
 	}
 
+	if spec.Features != nil && spec.Features.KubeStateMetricsCore != nil {
+		if spec.Features.KubeStateMetricsCore.Conf != nil {
+			if err = IsValidCustomConfigSpec(spec.Features.KubeStateMetricsCore.Conf); err != nil {
+				errs = append(errs, fmt.Errorf("invalid spec.features.kubeStateMetricsCore.conf, err: %v", err))
+			}
+		}
+	}
```

Comment:
```suggestion
	if spec.Features != nil && spec.Features.KubeStateMetricsCore != nil && spec.Features.KubeStateMetricsCore.Conf != nil {
		if err = IsValidCustomConfigSpec(spec.Features.KubeStateMetricsCore.Conf); err != nil {
			errs = append(errs, fmt.Errorf("invalid spec.features.kubeStateMetricsCore.conf, err: %v", err))
		}
	}
```

Reply:
so `spec.Features.KubeStateMetricsCore.Conf` can be nil and actually we rely on the fact that if it is not set it is nil to create the default config.

Reply:
edit: after discussing, we will instantiate the features field if nil and Default all features.